File: Уровень2/Уровень2Б/Задачи_на_множители/Простое_число.py
#https://algoprog.ru/material/p973

def is_prime(n):
  if n == 1:
    return False
  for i in range(2, int(n ** 0.5) + 1):
    if n % i == 0:
      return False
  return True


if __name__ == '__main__':
  k = int(input())
  cnt, j = 0, 2
  while True:
    if is_prime(j):
      cnt += 1
    if cnt == k:
      break
    j += 1
  print(j)

File: Уровень2/Уровень2Б/Задачи_на_множители/Простое_число.cpp
//https://algoprog.ru/material/p973

#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(int n) {
    if (n == 1)
        return false;
    int i = 2;
    while (i*i <= n) {
        if (n % i == 0)
            return false;
        i += 1;
    }
    return true;
}

int main() {
    int k;
    cin >> k;
    int cnt = 0, j = 2;
    while (true) {
        if (is_prime(j))
            cnt++;
        if (cnt == k)
            break;
        j++;
    }
    cout << j << endl;
    return 0;
}
